src/internal/connector/sharepoint/libraries_test.go

```diff
@@ -0,0 +1,313 @@
+package sharepoint_test
```

Comment:
not sure how I feel about duplicating so much code for tests. It seems like it would be difficult to keep them in sync which could lead to uneven testing in the long run

Reply:
I'm happy to consolidate them in a follow-up change.  How's that sound?